universal-application-tool-0.0.1/app/repository/ProgramRepository.java

```diff
@@ -47,6 +52,50 @@ public Program updateProgramSync(Program program) {
     return program;
   }
 
+  public void updateQuestionVersions(Program program) {
+    Transaction transaction = ebeanServer.beginTransaction();
+    try {
+      updateQuestionVersions(program, transaction);
+      ebeanServer.commitTransaction();
+    } finally {
+      ebeanServer.endTransaction();
+    }
+  }
+
+  public void updateQuestionVersions(Program program, Transaction transaction) {
+    ProgramDefinition.Builder updatedDefinition =
+        program.getProgramDefinition().toBuilder().setBlockDefinitions(ImmutableList.of());
+    for (BlockDefinition block : program.getProgramDefinition().blockDefinitions()) {
+      BlockDefinition.Builder updatedBlock =
+          block.toBuilder().setProgramQuestionDefinitions(ImmutableList.of());
+      for (ProgramQuestionDefinition question : block.programQuestionDefinitions()) {
+        Optional<Question> updatedQuestion =
+            ebeanServer
+                .find(Question.class)
+                .usingTransaction(transaction)
+                .where()
+                .eq(
+                    "name",
+                    ebeanServer
+                        .find(Question.class)
```

Comment:
Question (blocking): does this find also need `usingTransaction(transaction)` or will it inherit the transaction from the find it's nested in?

Reply:
I guess I don't see how it couldn't 🤔  resolving this